shared/components/Time.js

```diff
@@ -4,18 +4,60 @@ import Tooltip from 'components/Tooltip';
 import distanceInWordsToNow from 'date-fns/distance_in_words_to_now';
 import format from 'date-fns/format';
 
+let callbacks = [];
+
+// This is a shared timer that fires every second, used for
+// updating all Time components across the page all at once.
+setInterval(() => {
+  callbacks.forEach(cb => cb());
+}, 1000 * 60);
+
+function eachMinute(fn) {
+  callbacks.push(fn);
+
+  return () => {
+    callbacks = callbacks.filter(cb => cb !== fn);
+  }
+}
+
 type Props = {
   dateTime: string,
   children?: React.Node,
 };
 
-function Time({ dateTime, children }: Props) {
-  const date = new Date(dateTime);
-  return (
-    <Tooltip tooltip={format(date, 'MMMM Do, YYYY h:mm a')} placement="bottom">
-      <time dateTime={dateTime}>{children || distanceInWordsToNow(date)}</time>
-    </Tooltip>
-  );
+type State = {
+  autoUpdatingDateTime: string
+}
+
+class Time extends React.Component<Props, State> {
+  removeEachMinuteCallback: () => void;
+
+  state = {
+    autoUpdatingDateTime: this.props.dateTime
+  }
+
+  componentDidMount() {
+    this.removeEachMinuteCallback = eachMinute(() => {
+      this.setState({
+        autoUpdatingDateTime:
+          new Date(
+            Date.parse(this.state.autoUpdatingDateTime) + (1000 * 60)
```

Comment:
Well, we don't want to actually change the timestamp itself – otherwise it will keep moving forward as the component is mounted. We should want to make sure that the component re-renders so that `distanceInWordsToNow` is recalculated. 

I'd suggest changing `autoUpdatingDateTime` -> `key` and then each time the callback is triggered just increment `key` and ensure it's passed as the prop to `<time key={this.state.key}>` ?

Reply:
Ahh I see. Makes sense. We could also just do a `forceUpdate` every minute as well. That would eliminate the need for internal state altogether. What do you think?

Reply:
Sounds even better

Reply:
Cool, all set!